frame/collective/src/lib.rs

```diff
@@ -683,29 +521,197 @@ decl_module! {
 			}
 		}
 
-		/// Disapprove a proposal, close, and remove it from the system, regardless of its current state.
+		/// Disapprove a proposal, close, and remove it from the system,
+		/// regardless of its current state.
 		///
 		/// Must be called by the Root origin.
 		///
 		/// Parameters:
 		/// * `proposal_hash`: The hash of the proposal that should be disapproved.
-		///
-		/// # <weight>
-		/// Complexity: O(P) where P is the number of max proposals
-		/// DB Weight:
-		/// * Reads: Proposals
-		/// * Writes: Voting, Proposals, ProposalOf
-		/// # </weight>
-		#[weight = T::WeightInfo::disapprove_proposal(T::MaxProposals::get())]
-		fn disapprove_proposal(origin, proposal_hash: T::Hash) -> DispatchResultWithPostInfo {
+		#[pallet::weight(
+			T::WeightInfo::disapprove_proposal(T::MaxProposals::get())
+		)]
+		pub fn disapprove_proposal(
+			origin: OriginFor<T>,
+			proposal_hash: T::Hash
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let proposal_count = Self::do_disapprove_proposal(proposal_hash);
 			Ok(Some(T::WeightInfo::disapprove_proposal(proposal_count)).into())
 		}
 	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::metadata(T::AccountId = "AccountId", T::Hash = "Hash")]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		/// A motion (given hash) has been proposed (by given account) with a threshold (given
+		/// `MemberCount`).
+		/// \[account, proposal_index, proposal_hash, threshold\]
+		Proposed(T::AccountId, ProposalIndex, T::Hash, MemberCount),
+		/// A motion (given hash) has been voted on by given account, leaving
+		/// a tally (yes votes and no votes given respectively as `MemberCount`).
+		/// \[account, proposal_hash, voted, yes, no\]
+		Voted(T::AccountId, T::Hash, bool, MemberCount, MemberCount),
+		/// A motion was approved by the required threshold.
+		/// \[proposal_hash\]
+		Approved(T::Hash),
+		/// A motion was not approved by the required threshold.
+		/// \[proposal_hash\]
+		Disapproved(T::Hash),
+		/// A motion was executed; result will be `Ok` if it returned without error.
+		/// \[proposal_hash, result\]
+		Executed(T::Hash, DispatchResult),
+		/// A single member did some action; result will be `Ok` if it returned without error.
+		/// \[proposal_hash, result\]
+		MemberExecuted(T::Hash, DispatchResult),
+		/// A proposal was closed because its threshold was reached or after its duration was up.
+		/// \[proposal_hash, yes, no\]
+		Closed(T::Hash, MemberCount, MemberCount),
+	}
+
+	/// Old name generated by `decl_event`.
+	// TODO :: Have to re-check is "Note message" required ?
+	// #[deprecated(note = "use `Event` instead")]
```

Comment:
if we keep it I think we can deprecate it
```suggestion
	#[deprecated(note = "use `Event` instead")]
```